admin/src/services/api.js

```diff
@@ -209,11 +209,15 @@ class api {
 
   uploadFile(path, arr, properties) {
     const names = arr.map((e) => e.name || e);
-    const files = arr.filter((e) => typeof e === "object");
+    let files = arr.filter((e) => typeof e === "object");
     let formData = new FormData();
-    for (let i = 0; i < files.length; i++) {
-      const safeFilename = encodeURIComponent(files[i].name.replace(/'/g, ""));
-      formData.append(files[i].name, files[i], safeFilename);
+    for (let file of files) {
+      const safeFilename = encodeURIComponent(file.name.replaceAll(/'/g, ""));
+      Object.defineProperty(file, "name", {
+        writable: true,
+        value: safeFilename,
+      });
+      formData.append(file.name, file, safeFilename);
```

Comment:
Not mandatory to put filename (3rd argument), it uses automatically name from file if file is type File